File: src/chipschallenge/blockreactions/SlipReaction.java

package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.BlockContainerFullException;
import chipschallenge.Move.Moves;


public abstract class SlipReaction extends BlockReaction {

    @Override
    public abstract void react(Block moving, Block standing) throws BlockContainerFullException;

    @Override
    public abstract boolean canMove(Block moving, Block standing);

    @Override
    public final Moves causesSlip(Block moving, Block standing) {
        return moving.getFacing();
    }

}

File: ChipsChallenge.Shared/Blockreactions/SlipReaction.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    using Moves = Move.Moves;

    public abstract class SlipReaction : BlockReaction
    {
        public abstract override void React(Block moving, Block standing);

        public abstract override bool canMove(Block moving, Block standing);

        public override Moves? CausesSlip(Block moving, Block standing)
        {
            return moving.Facing;
        }
    }
}